Fix reboot_gnome on s390

OleksandrOrlov opened the pull request:
Previously test checked for 'shut off' in case of s390 reboot_gnome, but
in fact, the vm state was 'running', as vm parameters were redefined
during the initial reboot and reboot=>destroy was changed to
reboot=>undef.

The commit adds an additional verification if the reboot is performing
on s390 system and it is not an initial one.

1. Related ticket: [poo#36279](https://progress.opensuse.org/issues/36279)
2. Verification runs:
- reboot_gnome on s380: http://oorlov-vm.qa.suse.de/tests/55
- reboot_gnome on xen (to check, that my changes do not break tests on other environments): http://oorlov-vm.qa.suse.de/tests/53
- To check, that my changes don't break tests on s390, that use assert_shutdown in other modules (e.g. shutdown.pm): http://oorlov-vm.qa.suse.de/tests/55#step/shutdown/19

mnowaksuse commented on tests/x11/reboot_gnome.pm:
```diff
@@ -19,7 +19,7 @@ use utils;
     my ($self) = @_;
     # 'keepconsole => 1' is workaround for bsc#1044072
-    power_action('reboot', keepconsole => 1);
+    power_action('reboot', keepconsole => 1, s390_reboot => get_var('S390_ZKVM', 0));
```
Do we need the explicit `0` here? `power_action()` seems to gate it sufficiently.

SergioAtSUSE replied:
@mnowaksuse, this would allow us to control the behavior by setting, without changing code.

Or do you mean the exlicit 0 for get_var()?

mnowaksuse replied:
Sorry for not making it clear. Yes, I meant the `0` in `get_var('S390_ZKVM', 0)`.

rwx788 commented on tests/x11/reboot_gnome.pm:
```diff
@@ -19,7 +19,7 @@ use utils;
     my ($self) = @_;
     # 'keepconsole => 1' is workaround for bsc#1044072
-    power_action('reboot', keepconsole => 1);
+    power_action('reboot', keepconsole => 1, s390_reboot => get_var('S390_ZKVM', 0));
```
I would suggest something to explicitly emphasize that it's about any reboot except initial. E.g. reboot_installed_system_zkvm . Just s390 can be be misleading as includes zVM.

OleksandrOrlov commented:
Changed the behavior a bit. Now, 'first_reboot' variable is sent in 'reboot_after_installation' instead of specifying it explicitly for s380 in reboot_gnome. I like that approach more.

Also, updated the verification runs accordingly.

Please feel free to share your thoughts.
